tests/integrations/subprocess/test_Subprocess__stream_output.py

```diff
@@ -74,3 +84,14 @@ def test_process_exit_with_queued_output(mock_sub, popen_process, capsys):
 
     mock_sub.stream_output("testing", popen_process)
     assert capsys.readouterr().out == ("output line 1\n" "\n" "output line 3\n")
+    mock_sub.cleanup.assert_called_once_with("testing", popen_process)
+
+
+@pytest.mark.parametrize("stop_func_ret_val", (True, False))
+def test_stop_func(mock_sub, popen_process, stop_func_ret_val, capsys):
+    """All output is printed whether stop_func aborts streaming or not."""
+    mock_sub.stream_output(
+        "testing", popen_process, stop_func=lambda: stop_func_ret_val
+    )
+    assert capsys.readouterr().out == ("output line 1\n" "\n" "output line 3\n")
+    mock_sub.cleanup.assert_called_once_with("testing", popen_process)
```

Comment:
Given we are only mimicking a real process for testing, I couldn't really think of a a way to more meaningfully test `stop_func`. I am definitely open to ideas.